src/db.c

```diff
@@ -1075,7 +1075,9 @@ void shutdownCommand(client *c) {
     if (!(flags & SHUTDOWN_NOSAVE) && scriptIsTimedout()) {
         /* Script timed out. Shutdown allowed only with the NOSAVE flag. See
          * also processCommand where these errors are returned. */
-        if (scriptIsEval())
+        if (server.busy_module)
+            addReplyErrorObject(c, shared.slowmoduleerr);
```

Comment:
Modules are not always user-controlled, unlike scripts, so I think it may not always make sense to the user to get this error message. Instead, I think it might be useful for a module to control the reply in this case.

Reply:
ok, so it should be an argument to the RM_Yield API?